### DISCUSSION TITLE
When celery result contains Chinese fomat

### DISCUSSION THREAD

**Discussion by LuanShiZhiJian** [celery/celery#7021] [Q&A] [answered]
# Checklist

- [ ] I have verified that the issue exists against the `master` branch of Celery.
- [ ] This has already been asked to the [discussion group](https://groups.google.com/forum/#!forum/celery-users) first.
- [ ] I have read the relevant section in the
  [contribution guide](http://docs.celeryproject.org/en/latest/contributing.html#other-bugs)
  on reporting bugs.
- [ ] I have checked the [issues list](https://github.com/celery/celery/issues?q=is%3Aissue+label%3A%22Issue+Type%3A+Bug+Report%22+-label%3A%22Category%3A+Documentation%22)
  for similar or identical bug reports.
- [ ] I have checked the [pull requests list](https://github.com/celery/celery/pulls?q=is%3Apr+label%3A%22PR+Type%3A+Bugfix%22+-label%3A%22Category%3A+Documentation%22)
  for existing proposed fixes.
- [ ] I have checked the [commit log](https://github.com/celery/celery/commits/master)
  to find out if the bug was already fixed in the master branch.
- [ ] I have included all related issues and possible duplicate issues
  in this issue (If there are none, check this box anyway).

## Mandatory Debugging Information

- [ ] I have included the output of ``celery -A proj report`` in the issue.
    (if you are not able to do this, then at least specify the Celery
     version affected).
- [ ] I have verified that the issue exists against the `master` branch of Celery.
- [ ] I have included the contents of ``pip freeze`` in the issue.
- [ ] I have included all the versions of all the external dependencies required
  to reproduce this bug.

## Optional Debugging Information

- [ ] I have tried reproducing the issue on more than one Python version
  and/or implementation.
- [ ] I have tried reproducing the issue on more than one message broker and/or
  result backend.
- [ ] I have tried reproducing the issue on more than one version of the message
  broker and/or result backend.
- [ ] I have tried reproducing the issue on more than one operating system.
- [ ] I have tried reproducing the issue on more than one workers pool.
- [ ] I have tried reproducing the issue with autoscaling, retries,
  ETA/Countdown & rate limits disabled.
- [ ] I have tried reproducing the issue after downgrading
  and/or upgrading Celery and its dependencies.

#### Related Issues

- I used to Django Web and when celery result contains Chinese fomat,the Chinese characters in the SQL database are encoded in Unicode format, not UTF8,Please advice,thanks.
![image](https://user-images.githubusercontent.com/41558924/84249466-5e335a80-ab3d-11ea-9073-d7c822c83aac.png)

## Environment & Settings

**Celery version**: 4.4.5

<details>
<summary><b><code>celery report</code> Output:</b></summary>
<p>

```
-------------- celery@CN-GZWNH42 v4.4.5 (cliffs)
--- ***** -----
-- ******* ---- Windows-10-10.0.18362-SP0 2020-06-10 17:39:54
- *** --- * ---
- ** ---------- [config]
- ** ---------- .> app:         sms:0x157d51ba9c8
- ** ---------- .> transport:   redis://xxxxxxxxxxxx:6379/13
- ** ---------- .> results:
- *** --- * --- .> concurrency: 4 (eventlet)
-- ******* ---- .> task events: OFF (enable -E to monitor tasks in this worker)
--- ***** -----
 -------------- [queues]
                .> celery           exchange=celery(direct) key=celery


[tasks]
  . mycelery.sms.tasks.send_sms
  . mycelery.sms.tasks.send_sms2

[2020-06-10 17:39:55,655: INFO/MainProcess] Connected to redis://xxxxxxxxx:6379/13
[2020-06-10 17:39:55,689: INFO/MainProcess] mingle: searching for neighbors
[2020-06-10 17:39:58,855: INFO/MainProcess] mingle: all alone
[2020-06-10 17:39:59,824: INFO/MainProcess] celery@CN-GZWNH42 ready.
[2020-06-10 17:39:59,869: INFO/MainProcess] pidbox: Connected to redis://xxxxxxxxxxxx:6379/13.
[2020-06-10 17:40:09,649: INFO/MainProcess] Received task: mycelery.sms.tasks.send_sms2[9b686dc8-c
398-4c98-9284-cf91fec762a5]
[2020-06-10 17:40:09,651: WARNING/MainProcess] 向手机号110发送短信成功2!
[2020-06-10 17:40:10,953: INFO/MainProcess] Task mycelery.sms.tasks.send_sms2[9b686dc8-c398-4c98-9
284-cf91fec762a5] succeeded in 1.2959999999984575s: '向手机号110发送短信成功!!'
```

</p>
</details>

# Steps to Reproduce
#### Settings:
```
BROKER_URL = 'redis://xxxxxx:6379/13'
CELERY_RESULT_BACKEND = 'django-db'

CELERY_RESULT_SERIALIZER = 'json'
CELERY_TIMEZONE = 'Asia/Shanghai'
CELERY_TASK_SERIALIZER = 'json'
CELERY_ENABLE_UTC = True
# CELERY_ACCEPT_CONTENT = ["json"]
# CELERY_RESULT_SERIALIZER = 'json'
# CELERY_TASK_RESULT_EXPIRES = 60 * 60 * 24
# CELERY_ACCEPT_CONTENT = ["json"]
```

## Required Dependencies

* **Minimal Python Version**: 3.7.5
* **Minimal Celery Version**: 4.4.5
* **Minimal Kombu Version**:  4.6.10
* **Minimal Broker Version**: redis 3.5.3
* **Minimal Result Backend Version**: PyMySQL 0.9.3
* **Minimal OS and/or Kernel Version**: Window10
* **Minimal Broker Client Version**: Redis
* **Minimal Result Backend Client Version**: django-db

### Python Packages

<details>
<summary><b><code>pip freeze</code> Output:</b></summary>
<p>

</p>
</details>

### Other Dependencies

<details>
<p>
N/A
</p>
</details>

## Minimally Reproducible Test Case

<details>
<p>

task code：
```
@app.task  
def send_sms2(mobile):
    print("向手机号%s发送短信成功2!" % mobile)
    time.sleep(1)
    return "向手机号%s发送短信成功!!" % mobile
```

</p>
</details>

# Expected Behavior

Chinese used UTF-8 format to displayed in normally and the SQL database(may include others)

# Actual Behavior

Chinese used Unicode format to display

**Comment by auvipy** [comment]
update your reported issue please.

**Comment by LuanShiZhiJian** [comment]
> 请更新您报告的问题。

Too complicated to know what to update, what information do you need me to provide?Thanks

**Comment by LuanShiZhiJian** [comment]
> 请更新您报告的问题。

I updated celery report, is it helpful?

**Comment by auvipy** [comment]
fill all the parts of this issue.

**Comment by LuanShiZhiJian** [comment]
> fill all the parts of this issue.

Hello,I have no additional content,Please advice,thanks.

**Comment by auvipy** [comment]
I can see you commented out # CELERY_RESULT_SERIALIZER = 'json' , using that what you get? also, can you check using a more recent version to fix this for you? I need to make sure whether its a celery bug or not :)

**Comment by thedrow** [comment]
Can you paste a result as an example we can inspect?

**Comment by nianqinianyi** [comment, marked as answer]
I had the same issue, and I think I figured out after researching.
It appears that Celery uses the default parameters when calling `json.dumps`, which means that `ensure_ascii` is set to `True` by default.
I'm not sure if Celery has any parameters to set a different default value for `ensure_ascii`. 
However, I resolved this issue by customizing the JSON serializer.

```python
import json
import os

from celery import Celery
from kombu.serialization import register


def json_serializer(data):
    # 解决中文被unicode编码的问题
    return json.dumps(data, ensure_ascii=False)


def json_deserializer(data):
    return json.loads(data)


register('json', json_serializer, json_deserializer, 'application/json')

# 设置 Django 的默认设置模块
os.environ.setdefault('DJANGO_SETTINGS_MODULE', 'my_project.settings')

app = Celery('my_project')

# 使用与 Django 设置相同的配置文件
app.config_from_object('django.conf:settings', namespace='CELERY')
```
